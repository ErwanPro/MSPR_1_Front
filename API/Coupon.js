import Constants from "expo-constants";
if (process.env.NODE_ENV == "test") {
	SERVER_ADDR = process.env.SERVER_ADDR;
}else {
	SERVER_ADDR = Constants.manifest.extra.urlServer;
}
import axios from "react-native-axios";

/**
 * 
 * @param {*} id id du coupon dans la BDD
 * @returns L'ensemble des informations du coupon : id, libelle, pourcentage de réduction, date de validité
 */
export async function getCouponById(id) {
	console.log(`${SERVER_ADDR}/coupon/${id}`);
	return (await axios.get(`${SERVER_ADDR}/coupon/${id}`)).data;
}

/**
 * 
 * @returns L'ensemble des coupons présents dans la BDD de manière détaillée : id, libelle, pourcentage de réduction, date de validité
 */
export async function getCoupons() {
	try {
		return (await axios.get(`${SERVER_ADDR}/coupon`)).data;
	} catch (e) {
		console.error(e);
	}
}
