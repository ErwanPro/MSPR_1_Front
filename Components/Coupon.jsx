import React from "react";
import { Text, StyleSheet, View } from "react-native";

class Coupon extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		const { libelle, pourcentage, dateValidite } = this.props;
		return (
			<View style={styles.container}>
				<Text style={styles.title}>{libelle}</Text>
				<Text style={styles.title}>{pourcentage}%</Text>
				<Text style={styles.title}>{dateValidite}</Text>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: "#fff",
		alignItems: "center",
		justifyContent: "center",
		borderRadius: 20,
		margin: 15,
		padding: 10,
		paddingBottom: 15,
		elevation: 10,
		shadowColor: "#000",
		shadowOffset: {
			width: 0,
			height: 5,
		},
		shadowOpacity: 0.34,
		shadowRadius: 6.27,
	},
	title: {
		fontSize: 17,
		fontWeight: "bold",
		marginVertical: 15,
	},
	addressContainer: {
		alignItems: "center",
		justifyContent: "space-between",
		fontSize: 13,
		marginBottom: 12,
	},
	address: {
		fontSize: 13,
	},
});

export default Coupon;
