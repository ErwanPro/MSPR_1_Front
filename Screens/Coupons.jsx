import React from "react";
import { FlatList, RefreshControl } from "react-native";
import { connect } from "react-redux";
import { getCoupons } from "../API/Coupon";
import Coupon from "../Components/Coupon";

class Coupons extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			refreshing: false,
			coupons: [],
		};
	}

	async componentDidMount() {
		// const coupons = await getCoupons();
		// this.setState({ coupons: coupons });
	}

	onRefresh = async () => {
		const coupons = await getCoupons();
		this.setState({ coupons: coupons, refreshing: false });
	};

	render() {
		console.log(this.props.coupons);
		return (
			<FlatList
				style={{ backgroundColor: "white" }}
				data={this.props.coupons}
				renderItem={({ item, index }) => <Coupon {...item} />}
				refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.onRefresh} />}
				keyExtractor={item => item._id}
			/>
		);
	}
}

const mapStateToProps = state => {
	return {
		coupons: state.coupons,
	};
};

export default connect(mapStateToProps)(Coupons);
