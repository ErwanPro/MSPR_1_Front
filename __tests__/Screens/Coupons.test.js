import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";
import { Coupons } from "../../Screens/Coupons.jsx";
import Coupon from "../../Components/Coupon.jsx";

describe("<Coupons/>", () => {
	describe("without Coupons", () => {
		it("should have 0 child", () => {
			const couponsInstance = renderer.create(<Coupons />).root;
			const children = couponsInstance.findByType(View).children.child;
			expect(children).toBeFalsy();
		});
	});

	describe("With Coupons", () => {
		let couponsInstance;
		beforeAll(() => {
			const coupons = [
				{
					_id: "603f51e40f03e30fb0453f85",
					libelle: "Chaussures en promo",
					pourcentage: "20",
					dateValidite: "10/03/2021",
					__v: 0,
				},
				{
					_id: "603f51fa0f03e30fb0453f86",
					libelle: "Tshirts en promo",
					pourcentage: "35",
					dateValidite: "20/03/2021",
					__v: 0,
				},
				{
					_id: "603f521a0f03e30fb0453f87",
					libelle: "Pantalon en promo",
					pourcentage: "15",
					dateValidite: "18/03/2021",
					__v: 0,
				},
			];
			 couponsInstance = renderer.create(<Coupons coupons={coupons} />);
		});
		it("should have 3 child", () => {
			expect(couponsInstance.root.findAllByType(Coupon).length).toBe(3);
		});
		it("Should render properly", () => {
			expect(couponsInstance.toJSON()).toMatchSnapshot();
		});
	});
});
