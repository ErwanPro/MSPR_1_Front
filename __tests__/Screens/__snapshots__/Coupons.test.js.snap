// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coupons/> With Coupons Should render properly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "__v": 0,
        "_id": "603f51e40f03e30fb0453f85",
        "dateValidite": "10/03/2021",
        "libelle": "Chaussures en promo",
        "pourcentage": "20",
      },
      Object {
        "__v": 0,
        "_id": "603f51fa0f03e30fb0453f86",
        "dateValidite": "20/03/2021",
        "libelle": "Tshirts en promo",
        "pourcentage": "35",
      },
      Object {
        "__v": 0,
        "_id": "603f521a0f03e30fb0453f87",
        "dateValidite": "18/03/2021",
        "libelle": "Pantalon en promo",
        "pourcentage": "15",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "elevation": 10,
            "flex": 1,
            "flexDirection": "row",
            "margin": 15,
            "padding": 10,
            "paddingBottom": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 15,
              }
            }
          >
            20
            %
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "height": "100%",
              "justifyContent": "space-between",
              "marginTop": 7,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Chaussures en promo
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            expire le 
            10/03/2021
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "elevation": 10,
            "flex": 1,
            "flexDirection": "row",
            "margin": 15,
            "padding": 10,
            "paddingBottom": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 15,
              }
            }
          >
            35
            %
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "height": "100%",
              "justifyContent": "space-between",
              "marginTop": 7,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Tshirts en promo
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            expire le 
            20/03/2021
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "elevation": 10,
            "flex": 1,
            "flexDirection": "row",
            "margin": 15,
            "padding": 10,
            "paddingBottom": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 15,
              }
            }
          >
            15
            %
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "height": "100%",
              "justifyContent": "space-between",
              "marginTop": 7,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Pantalon en promo
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            expire le 
            18/03/2021
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
